File: Linked List/singly/updation/list.java
// Singly Linked List - Updation (Java)
// Demonstrates updating the first node matching a target value.

class Node {
    int data;
    Node next;
    Node(int value) { this.data = value; this.next = null; }
}

public class list {
    static boolean updateFirst(Node head, int target, int newValue) {
        Node current = head;
        while (current != null) {
            if (current.data == target) {
                current.data = newValue;
                return true;
            }
            current = current.next;
        }
        return false;
    }

    static void traverse(Node head) {
        Node cur = head;
        while (cur != null) {
            System.out.print(cur.data + " ");
            cur = cur.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        // Build list: 10 -> 20 -> 30
        Node head = new Node(10);
        head.next = new Node(20);
        head.next.next = new Node(30);

        System.out.print("Before update: ");
        traverse(head);

        boolean updated = updateFirst(head, 20, 25);
        System.out.println(updated ? "Updated successfully" : "Value not found");

        System.out.print("After update: ");
        traverse(head);
    }
}

File: Linked List/singly/updation/list.py
# Singly Linked List - Updation (Python)
# Demonstrates updating the first occurrence of a target value.

class Node:
    def __init__(self, value: int) -> None:
        self.data = value
        self.next = None


def update_first(head: "Node | None", target: int, new_value: int) -> bool:
    current = head
    while current is not None:
        if current.data == target:
            current.data = new_value
            return True
        current = current.next
    return False


def traverse(head: "Node | None") -> None:
    cur = head
    while cur is not None:
        print(cur.data, end=" ")
        cur = cur.next
    print()


if __name__ == "__main__":
    # Build list: 10 -> 20 -> 30
    head = Node(10)
    head.next = Node(20)
    head.next.next = Node(30)

    print("Before update:", end=" ")
    traverse(head)

    updated = update_first(head, 20, 25)
    print("Updated successfully" if updated else "Value not found")

    print("After update:", end=" ")
    traverse(head)


